File: mobile/trace/lib/features/class_management/domain/repositories/class_repository.dart
import '../entities/class_entity.dart';
import '../entities/class_statistics.dart';
import '../../../authentication/domain/entities/user_entity.dart';

abstract class ClassRepository {
  Future<List<ClassEntity>> getLecturerClasses();
  Future<List<ClassEntity>> getEnrolledClasses();
  Future<List<ClassEntity>> searchClasses(String query);
  Future<ClassEntity?> getClassDetails(String classId);
  Future<ClassStatistics?> getClassStatistics(String classId, UserEntity user);
  Future<void> createClass(ClassEntity classModel);
  Future<void> updateClass(String classId, ClassEntity classModel);
  Future<void> deleteClass(String classId);
  Future<void> enrollStudent(String classId, String studentId);
  Future<void> unenrollStudent(String classId, String studentId);
}

File: mobile/trace/domain/repositories/ClassRepository.ts
import { Class } from '@/domain/entities/Class';
import { AppError } from '@/shared/errors/AppError';

export interface ClassRepository {
    getClass(id: string): Promise<Class | AppError>;
    getClasses(): Promise<Class[] | AppError>;
    createClass(data: Omit<Class, 'id' | 'createdAt' | 'updatedAt'>): Promise<Class | AppError>;
    updateClass(id: string, data: Partial<Class>): Promise<Class | AppError>;
    deleteClass(id: string): Promise<void | AppError>;
    addStudent(classId: string, studentId: string): Promise<void | AppError>;
    removeStudent(classId: string, studentId: string): Promise<void | AppError>;
    getLecturerClasses(lecturerId: string): Promise<Class[] | AppError>;
    getStudentClasses(studentId: string): Promise<Class[] | AppError>;
    searchClasses(query: string): Promise<Class[] | AppError>;
} 